src/integrationTest/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerIssueFinalDecisionIT.java

```diff
@@ -0,0 +1,377 @@
+import static uk.gov.hmcts.sptribs.ciccase.model.ContactPreferenceType.EMAIL;
+import static uk.gov.hmcts.sptribs.ciccase.model.RepresentativeCIC.REPRESENTATIVE;
+import static uk.gov.hmcts.sptribs.ciccase.model.RespondentCIC.RESPONDENT;
+import static uk.gov.hmcts.sptribs.ciccase.model.SubjectCIC.SUBJECT;
+import static uk.gov.hmcts.sptribs.document.DocumentConstants.FINAL_DECISION_ANNEX_TEMPLATE_ID;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.ABOUT_TO_START_URL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.ABOUT_TO_SUBMIT_URL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.AUTHORIZATION;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.ISSUE_FINAL_DECISION_MID_EVENT_URL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.ISSUE_FINAL_DECISION_UPLOAD_MID_EVENT_URL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.SUBMITTED_URL;
+import static uk.gov.hmcts.sptribs.testutil.TestConstants.TEST_AUTHORIZATION_TOKEN;
+import static uk.gov.hmcts.sptribs.testutil.TestDataHelper.callbackRequest;
+import static uk.gov.hmcts.sptribs.testutil.TestDataHelper.caseData;
+import static uk.gov.hmcts.sptribs.testutil.TestEventConstants.CASEWORKER_ISSUE_FINAL_DECISION;
+import static uk.gov.hmcts.sptribs.testutil.TestResourceUtil.expectedResponse;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureMockMvc
+@ContextConfiguration(initializers = {IdamWireMock.PropertiesInitializer.class})
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class CaseworkerIssueFinalDecisionIT {
+    private static final String CASEWORKER_ISSUE_FINAL_DECISION_MID_EVENT_RESPONSE =
+        "classpath:responses/caseworker-issue-final-decision-mid-event-response.json";
+    private static final String CASEWORKER_ISSUE_FINAL_DECISION_UPLOAD_MID_EVENT_RESPONSE =
+        "classpath:responses/caseworker-issue-final-decision-upload-mid-event-response.json";
+    private static final String CASEWORKER_ISSUE_FINAL_DECISION_ABOUT_TO_SUBMIT_RESPONSE =
+        "classpath:responses/caseworker-issue-final-decision-about-to-submit-response.json";
+    private static final String CONFIRMATION_HEADER = "$.confirmation_header";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private WebMvcConfig webMvcConfig;
+
+    @MockBean
+    private IdamService idamService;
+
+    @MockBean
+    private CaseDataDocumentService caseDataDocumentService;
+
+    @MockBean
+    private NotificationServiceCIC notificationServiceCIC;
+
+    @BeforeAll
+    static void setUp() {
+        IdamWireMock.start();
+    }
+
+    @AfterAll
+    static void tearDown() {
+        IdamWireMock.stopAndReset();
+    }
+
+    @Test
+    void shouldSetDecisionSignatureInAboutToStart() throws Exception {
```

Comment:
Maybe the name of the test should emphasise that the decision signature is reset or set to an empty string?